docs/plugins/integrations/authentication.rst

```diff
@@ -0,0 +1,750 @@
+
+.. code-block:: php
+
+    <?php
+    use kamermans\OAuth2\Persistence\TokenPersistenceInterface;
+    use MauticPlugin\HelloWorldBundle\Integration\HelloWorldIntegration;
+    use MauticPlugin\IntegrationsBundle\Auth\Provider\Oauth2TwoLegged\Credentials\ClientCredentialsGrantInterface;
+    use MauticPlugin\IntegrationsBundle\Auth\Provider\Oauth2TwoLegged\Credentials\ScopeInterface;
+    use MauticPlugin\IntegrationsBundle\Auth\Provider\Oauth2TwoLegged\HttpFactory;
+    use MauticPlugin\IntegrationsBundle\Auth\Support\Oauth2\ConfigAccess\ConfigTokenPersistenceInterface;
+    use MauticPlugin\IntegrationsBundle\Helper\IntegrationsHelper;
+
+    /** @var $integrationsHelper IntegrationsHelper */
+    $integration = $integrationsHelper->getIntegration(HelloWorldIntegration::NAME);
+
+    $configuration = $integration->getIntegrationConfiguration();
+    $apiKeys       = $configuration->getApiKeys();
+
+    $credentials = new class(
+        'https://example.com/api/oauth/token',
+        'scope1,scope2',
+        $apiKeys['client_id'],
+        $apiKeys['client_secret']
+    ) implements ClientCredentialsGrantInterface, ScopeInterface {
+        private $authorizeUrl;
+        private $scope;
+        private $clientId;
+        private $clientSecret;
+
+        public function getAuthorizationUrl(): string
+        {
+            return $this->authorizeUrl;
+        }
+
+        public function getClientId(): ?string
+        {
+            return $this->clientId;
+        }
+
+        public function getClientSecret(): ?string
+        {
+            return $this->clientSecret;
+        }
+
+        public function getScope(): ?string
+        {
+            return $this->scope;
+        }
+    };
+
+    /** @var $tokenPersistenceFactory TokenPersistenceFactory */
+    $tokenPersistence = $tokenPersistenceFactory->create($integration);
+    $config           = new class($tokenPersistence) implements ConfigTokenPersistenceInterface {
+        private $tokenPersistence;
+
+        public function __construct(TokenPersistenceInterface$tokenPersistence)
+        {
+            $this->tokenPersistence = $tokenPersistence;
+        }
+
+        public function getTokenPersistence(): TokenPersistenceInterface
+        {
+            return $this->tokenPersistence;
+        }
+    };
+
+    /** @var $factory HttpFactory */
+    $client   = $factory->getClient($credentials, $config);
+    $response = $client->get('https://example.com/api/fetch');
+
+OAuth2 three legged
+===================
+
+Three legged OAuth2 with the code grant is the most complex to implement because it involves redirecting the user to the third party service to authenticate then sent back to Mautic to initiate the access token process using a code returned in the request.
```

Comment:
```suggestion
Three legged OAuth2 with the code grant is the most complex to implement because it involves redirecting the User to the third party service to authenticate then sent back to Mautic to initiate the access token process using a code returned in the request.
```